VerilogCodingStyle.md

```diff
@@ -797,6 +797,92 @@ endmodule : foo
 localparam bit ValBaz = 1;  // This comment describes the item to the left.
 ```
 
+It can sometimes be useful to structure the code using header-style comments in
+order to separate different functional parts (like FSMs, the main datapath or
+registers) within a module. In that case, the preferred style is as follows:
+
+```systemverilog
+module foo;
+
+  ////////////////
+  // Controller //
+  ////////////////
+  ...
+
+  ///////////////////////
+  // Main ALU Datapath //
+  ///////////////////////
+  ...
+
+  ///////////////////////////////////////////////////////
+  // Complicated part of the design that warrants some //
+  // more description in the header comment            //
+  ///////////////////////////////////////////////////////
+  ...
+
+  ///////////////////////////////////////////////////
```

Comment:
Will we allow the mismatch of length in this line and the actual heading sentence?

Reply:
I would say yes because then you can play with the separator width, even if the text inside the heading is super short.

Reply:
So.. For instance, this is allowed.

```
////////////////////////////////////////////
// It is ok to omit slashes
////////////////////////////////////////////
```

Reply:
yes this would be allowed

Reply:
Ok I remove this as well to standardize it as much as possible.